drivers/ds3231/ds3231.c

```diff
@@ -236,6 +257,158 @@ int ds3231_set_time(const ds3231_t *dev, const struct tm *time)
     return 0;
 }
 
+#if IS_USED(MODULE_DS3231_INT)
+int ds3231_await_alarm(ds3231_t *dev)
+{
+    mutex_t mutex = MUTEX_INIT_LOCKED;
+
+    assert(dev != NULL);
+    assert(gpio_is_valid(dev->int_pin));
+
+    if (gpio_init_int(dev->int_pin, GPIO_IN, GPIO_FALLING, _unlock, &mutex) < 0) {
+            return -EIO;
+    }
+
+    /* wait for alarm */
+    mutex_lock(&mutex);
+
+    uint8_t status, tmp;
+    _read(dev, REG_STATUS, &status, 1, 1, 0);
+
+    /* clear interrupt flags */
+    tmp = status & ~(STAT_A1F | STAT_A2F);
+    if (_write(dev, REG_STATUS, &tmp, 1, 0, 1) < 0) {
+        return -EIO;
+    }
+
+    return status & (STAT_A1F | STAT_A2F);
+}
+#endif
+
+int ds3231_set_alarm_1(const ds3231_t *dev, struct tm *time,
+                       ds3231_alm_1_mode_t trigger)
```

Comment:
```suggestion
typedef enum {
    DS3231_ALARM1,
    DS3231_ALARM2,
} ds3231_alarm_slot_t;

int ds3231_set_alarm(const ds3231_t *dev, ds3231_alarm_slot_t slot, struct tm *time,
                     ds3231_alm_1_mode_t trigger)
```

I think you can share some code here.

Reply:
It would be nice but the triggers are different for the two alarms (see ds3231_alm_1_mode_t and ds3231_alm_2_mode_t typedef). I'm open to a different solution if someone finds one.